docs/design-documents/20230228-connector-lifecycle-methods.md

```diff
@@ -0,0 +1,226 @@
+# Connector Lifecycle Methods
+
+## Introduction
+
+Connector Lifecycle Methods are additional integration points that would let
+connectors execute certain operations when the connector is first created, when
+its configuration is updated or when it gets deleted.
+
+### Goals
+
+- Lifecycle methods should be optional to implement.
+- The introduction of lifecycle methods must be backwards compatible with
+  existing connectors.
+- The connector needs to be notified when it was first created*.
+- The connector needs to be notified when its configuration has changed*.
+- The connector needs to be notified when it gets deleted.
+
+\*Actually when the pipeline is first started after that happened.
```

Comment:
Fully started or when Conduit attempted to start it?

Reply:
Why do I not understand this sentence 😅

Reply:
> Why do I not understand this sentence sweat_smile

Here's how I understood it: if a connector was created, but the pipeline has not been started, then the `OnCreate` event won't get triggered.

Reply:
oh I see! then I think we need some rephrasing here to make it clearer

Reply:
it does, but it was my first time reading the doc and this section comes before any explanation, I just think it needs some rephrasing to make it clearer

Reply:
looks good!